src/mlpack/methods/preprocess/preprocess_string_util.cpp

```diff
@@ -0,0 +1,156 @@
+/**
+ * @file preprocess_string_util.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to encode string dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include "mlpack/methods/preprocess/preprocess_string_util.hpp"
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Function neccessary to create a vector<vector<string>> by readin
+ * the contents of a file.
+ *
+ * @param filename Name of the file whose contents need to be preproccessed.
+ * @param columnDelimiter Delimiter used to split the columns of file.
+ */
+vector<vector<string>> CreateDataset(const string& filename,
+                                     char columnDelimiter)
+{
+  vector<vector<string>> dataset;
+  // Extracting the contents of file.
+  // File stream.
+  ifstream fin(filename);
+  if (!fin.is_open())
+    Log::Fatal << "Unable to open input file" << endl;
```

Comment:
It could be nice to check the `ifstream` failure to provide a more descriptive error message, do you think you could add that? :)